src/mlpack/methods/ann/rbm/spike_slab_rbm_policy_impl.hpp

```diff
@@ -0,0 +1,341 @@
+
+#ifndef MLPACK_METHODS_ANN_RBM_SPIKE_SLAB_RBM_POLICY_IMPL_HPP
+#define MLPACK_METHODS_ANN_RBM_SPIKE_SLAB_RBM_POLICY_IMPL_HPP
+
+#include "spike_slab_rbm_policy.hpp"
+
+namespace mlpack {
+namespace ann {
+
+inline SpikeSlabRBMPolicy::SpikeSlabRBMPolicy(const size_t visibleSize,
+      const size_t hiddenSize,
+      const size_t poolSize,
+      const double slabPenalty, 
+      double radius):
+      visibleSize(visibleSize),
+      hiddenSize(hiddenSize),
+      poolSize(poolSize),
+      slabPenalty(slabPenalty),
+      radius(radius)
+{
+  parameter.set_size(visibleSize * hiddenSize * poolSize +
+      visibleSize + hiddenSize);
+
+  assert(slabPenalty.n_rows == 1);
+  assert(slabPenalty.n_cols == 1);
+
+  visibleMean.set_size(visibleSize, 1);
+  spikeMean.set_size(hiddenSize, 1);
+  spikeSamples.set_size(hiddenSize, 1);
+  slabMean.set_size(poolSize, hiddenSize);
+};
+
+// Reset function
+inline void SpikeSlabRBMPolicy::Reset()
+{
+  // Weight shape D * K * N
+  weight = arma::cube(parameter.memptr(), visibleSize, poolSize, hiddenSize,
+      false, false);
+
+  // spike bias shape N * 1
+  spikeBias = arma::mat(parameter.memptr() + weight.n_elem, hiddenSize, 1,
+      false, false);
+
+  // visible penalty 1 * 1 => D * D(when used)
+  visiblePenalty = arma::mat(parameter.memptr() + weight.n_elem +
+      spikeBias.n_elem, 1, 1, false, false);
+
+  diagInvSlabPenalty = 1.0 / slabPenalty * arma::eye(poolSize, poolSize);
+  diagSlabPenalty = slabPenalty * arma::eye(poolSize, poolSize);
```

Comment:
Well, most of the calculations would not feasible if we change the `slabPenalty` to a scalar.
like P(s | v, h). if `slabPenalty` is scalar it would require the mean to scalar as well.
But the mean is given by `spike(i) * slabPenalty * weight.slice(i).t() * visible` here if slabPenalty is a scalar then the mean would not be a scalar or `1*1` matrix